drivers/devmapper/deviceset.go

```diff
@@ -2445,8 +2445,10 @@ func (devices *DeviceSet) UnmountDevice(hash, mountPath string) error {
 	defer devices.Unlock()
 
 	logrus.Debugf("devmapper: Unmount(%s)", mountPath)
-	if err := mount.Unmount(mountPath); err != nil {
-		return err
+	if ok, _ := Mounted(mountPath); ok {
+		if err := mount.Unmount(mountPath); err != nil {
+			return err
+		}
```

Comment:
this introduces a race.

Let's do the opposite, first we try to unmount and in case of errors check if it is mounted (if it is mounted then return the Unmount error)

Reply:
Obviously, your proposal is better, and I've changed it as you suggested.